### ISSUE TITLE
[Device request]: MHCozy 4-way dry relay wifi switch

### ISSUE THREAD

**Issue by MattiasNyholm** [athombv/com.tuya#118]
### Device information

[https://www.amazon.com/MHCOZY-Channel-Bluetooth-Wireless-Contact/dp/B09M6HTDVS/](https://www.amazon.com/MHCOZY-Channel-Bluetooth-Wireless-Contact/dp/B09M6HTDVS/)
The device specification is from a device configured in interlock mode, meaning that as one of the four relays are enabled the other three are automatically deactivated. It's possible to configure the relay in two other modes as well. 

This device is based on a hardware module that is available in multiple different physical formats, with wifi or zigbee, with two or four relays etc. 

I've translated all "name" fields from Chineese to english using Google translate. 

### Device specification

{ "device": { "active_time": 1722843748, "biz_type": 0, "category": "tdq", "create_time": 1722843748, "icon": "smart/icon/ay1528249126144ckqum/81c9c2fc51a0d416cd9977c29f08d271.png", "id": "bf3e5854d8d47fcfee3pme", "ip": "<redacted>", "lat": "<redacted>", "local_key": "<redacted>", "lon": "<redacted>", "model": "4-274", "name": "Pool pumpkontroll", "online": true, "owner_id": "<redacted>", "product_id": "n0kphfrgn3fd0iat", "product_name": "4 way switch", "status": [ { "code": "switch_1", "value": false }, { "code": "switch_2", "value": false }, { "code": "switch_3", "value": false }, { "code": "switch_4", "value": false }, { "code": "countdown_1", "value": 0 }, { "code": "countdown_2", "value": 0 }, { "code": "countdown_3", "value": 0 }, { "code": "countdown_4", "value": 0 }, { "code": "relay_status", "value": "0" }, { "code": "random_time", "value": "" }, { "code": "cycle_time", "value": "" }, { "code": "switch_inching", "value": "AAACAgACBAACBgAC" }, { "code": "switch_interlock", "value": "AA8AAAAAAAA=" } ], "sub": false, "time_zone": "+02:00", "uid": "<redacted>", "update_time": 1722843887, "uuid": "<redacted>" }, "specifications": { "category": "tdq", "functions": [ { "code": "switch_1", "desc": "{}", "name": "switch1", "type": "Boolean", "values": "{}" }, { "code": "switch_2", "desc": "{}", "name": "switch2", "type": "Boolean", "values": "{}" }, { "code": "switch_3", "desc": "{}", "name": "switch3", "type": "Boolean", "values": "{}" }, { "code": "switch_4", "desc": "{}", "name": "switch4", "type": "Boolean", "values": "{}" }, { "code": "countdown_1", "desc": "{\"unit\":\"s\",\"min\":0,\"max\":86400,\"scale\":0,\"step\":1}", "name": "Switch 1 countdown", "type": "Integer", "values": "{\"unit\":\"s\",\"min\":0,\"max\":86400,\"scale\":0,\"step\":1}" }, { "code": "countdown_2", "desc": "{\"unit\":\"s\",\"min\":0,\"max\":86400,\"scale\":0,\"step\":1}", "name": "Switch 2 countdown", "type": "Integer", "values": "{\"unit\":\"s\",\"min\":0,\"max\":86400,\"scale\":0,\"step\":1}" }, { "code": "countdown_3", "desc": "{\"unit\":\"s\",\"min\":0,\"max\":86400,\"scale\":0,\"step\":1}", "name": "Switch 3 countdown", "type": "Integer", "values": "{\"unit\":\"s\",\"min\":0,\"max\":86400,\"scale\":0,\"step\":1}" }, { "code": "countdown_4", "desc": "{\"unit\":\"s\",\"min\":0,\"max\":86400,\"scale\":0,\"step\":1}", "name": "switch 4 countdown", "type": "Integer", "values": "{\"unit\":\"s\",\"min\":0,\"max\":86400,\"scale\":0,\"step\":1}" }, { "code": "relay_status", "desc": "{\"range\":[\"0\",\"1\",\"2\"]}", "name": "Power-on status settings", "type": "Enum", "values": "{\"range\":[\"0\",\"1\",\"2\"]}" }, { "code": "random_time", "desc": "{\"maxlen\":255}", "name": "
Random timing", "type": "String", "values": "{\"maxlen\":255}" }, { "code": "cycle_time", "desc": "{\"maxlen\":255}", "name": "Loop timing", "type": "String", "values": "{\"maxlen\":255}" }, { "code": "switch_inching", "desc": "{\"maxlen\":255}", "name": "
Touch switch", "type": "String", "values": "{\"maxlen\":255}" }, { "code": "switch_interlock", "desc": "{}", "name": "Switch interlock", "type": "Raw", "values": "{}" } ], "status": [ { "code": "switch_1", "name": "switch 1", "type": "Boolean", "values": "{}" }, { "code": "switch_2", "name": "switch 2", "type": "Boolean", "values": "{}" }, { "code": "switch_3", "name": "switch 3", "type": "Boolean", "values": "{}" }, { "code": "switch_4", "name": "switch 4", "type": "Boolean", "values": "{}" }, { "code": "countdown_1", "name": "
Switch 1 countdown", "type": "Integer", "values": "{\"unit\":\"s\",\"min\":0,\"max\":86400,\"scale\":0,\"step\":1}" }, { "code": "countdown_2", "name": "Switch 2 countdown", "type": "Integer", "values": "{\"unit\":\"s\",\"min\":0,\"max\":86400,\"scale\":0,\"step\":1}" }, { "code": "countdown_3", "name": "Switch 3 countdown", "type": "Integer", "values": "{\"unit\":\"s\",\"min\":0,\"max\":86400,\"scale\":0,\"step\":1}" }, { "code": "countdown_4", "name": "Switch 4 countdown", "type": "Integer", "values": "{\"unit\":\"s\",\"min\":0,\"max\":86400,\"scale\":0,\"step\":1}" }, { "code": "relay_status", "name": "Power-on status settings", "type": "Enum", "values": "{\"range\":[\"0\",\"1\",\"2\"]}" }, { "code": "random_time", "name": "Random timing", "type": "String", "values": "{\"maxlen\":255}" }, { "code": "cycle_time", "name": "Loop timing", "type": "String", "values": "{\"maxlen\":255}" }, { "code": "switch_inching", "name": "Touch switch", "type": "String", "values": "{\"maxlen\":255}" }, { "code": "switch_interlock", "name": "Switch interlock", "type": "Raw", "values": "{}" } ] } }

**Comment by bobvandevijver** [comment]
Yes, it should indeed. I'm closing this now, version 1.2.0 which includes this should hit testing this week.

**Comment by bobvandevijver** [comment]
You will probably need to wait for 1.2.2, which contains a fix for the `relay_status` used by this device.

**Comment by MattiasNyholm** [comment]
> You will probably need to wait for 1.2.2, which contains a fix for the `relay_status` used by this device.

I just tested and controlling all four switches work just fine from Homey. With the interlock setting configured in Tuya I can turn e g switch 2 on and then the others will turn off automatically. 

Homey doesn’t detect changes eg in the Tuya app, but I guess that’s the relay status issue you referred to? I can test again when the fix is out.
